### PR TITLE
Remove connectionInfo and allow cloud storage on public repos

### PR DESCRIPTION
**Pull request by hannahlspencer**
### What does this PR do?
- Removes connectionInfo and replaces it with data source information from the API
- Adds functionality for public data sources to be used with cloud storage

This code will change with the introduction of the storage key being used in the backend.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### CODE REVIEW

File: client/src/pages/recording-view/components/plugins-pane.tsx [hunk 1]
```diff
@@ -44,10 +45,15 @@ export const PluginsPane = () => {
   const [modalSamples, setModalSamples] = useState<Float32Array>(new Float32Array([]));
   const [modalSpectrogram, setmodalSpectrogram] = useState(null);
   const [useCloudStorage, setUseCloudStorage] = useState(true);
-  const { dataSourcesQuery } = useUserSettings();
-  const connectionInfo = dataSourcesQuery?.data[`${account}/${container}`];
+  const dataSource = getDataSource(CLIENT_TYPE_API, account, container, true).data;
```
Comment (Nepomuceno) [inline client/src/pages/recording-view/components/plugins-pane.tsx:48]: We need to be careful this page might still be used with a blob datasource or with a local file so pining up this here might not be a good idea

Comment (Jade-Codes) [inline client/src/pages/recording-view/components/plugins-pane.tsx:48]: I think:

const {data }  = getDataSource(CLIENT_TYPE_API, account, container, true);

Is usually the preferred way of retrieving a property :)

File: client/src/pages/recording-view/components/plugins-pane.tsx [hunk 2]
```diff
@@ -305,7 +335,7 @@ export const PluginsPane = () => {
           ))}
         </select>
       </label>
-      {connectionInfo && (
+      {dataSource && (
```
Comment (Nepomuceno) [inline client/src/pages/recording-view/components/plugins-pane.tsx:338]: We should probably do a different check here for exemole you don't want that for local files but you do want this for personal blob storage

**Review by Nepomuceno — COMMENTED** [review]
We need to make sure it still works for local files and for local blob storage.

File: client/src/pages/recording-view/components/plugins-pane.tsx [hunk 3]
```diff
@@ -96,15 +102,39 @@ export const PluginsPane = () => {
...
+          },
+        ],
+        custom_params: {
+          start_freq: annotation ? annotation['core:freq_lower_edge'] : null,
+          end_freq: annotation ? annotation['core:freq_upper_edge'] : null,
+        },
+      };
+    }
+
+    if (useCloudStorage && dataSource) {
```
Comment (Jade-Codes) [inline client/src/pages/recording-view/components/plugins-pane.tsx:129]: I would just do:

if (useCloudStorage) 

once.

change sas_token to be null for now, then when the implementation for 'get_blob_token' is done, tie that in

File: client/src/pages/recording-view/components/plugins-pane.tsx (hunk 1 above)
Comment (hannahlspencer) [reply client/src/pages/recording-view/components/plugins-pane.tsx:48]: To get the sas token (for now)

Comment (Jade-Codes) [reply client/src/pages/recording-view/components/plugins-pane.tsx]: I would remove as we really shouldn't be using any form of sas token from the data sources or connection info going forward... Not sure why we are returning this from the API

**Pull request closed without merging**